File: abc213d.cpp
#include <bits/stdc++.h>
using namespace std;

vector<vector<int>> G;
vector<bool> visited;
vector<int> ans;

void dfs(int v, int p = -1){
    visited[v] = true;
    ans.push_back(v+1);
    for(auto &u : G[v]){
        if(u == p || visited[u]) continue;
        dfs(u, v);
        ans.push_back(v+1);
    }
}

int main(){
    int N; cin >> N;
    G.resize(N);
    visited.assign(N, false);

    for(int i=0; i<N-1; i++){
        int A, B; cin >> A >> B;
        A--; B--;
        G[A].push_back(B);
        G[B].push_back(A);
    }

    for(int i=0; i<N; i++){
        sort(G[i].begin(), G[i].end());
    }

    dfs(0);
    
    for(int i=0; i<ans.size(); i++){
        if(i > 0) cout << " ";
        cout << ans[i];
    }
    cout << endl;
}

File: abc213d.php
<?php

fscanf(STDIN, "%d\n", $N);
$G = array_fill(0, $N, []);
$visited = array_fill(0, $N, false);
$ans = [];

for($i=0; $i<$N-1; $i++){
    fscanf(STDIN, "%d %d\n", $A, $B);
    $A--; $B--;
    $G[$A][] = $B;
    $G[$B][] = $A;
}

foreach($G as $key => $value){
    sort($G[$key]);
}

dfs(0, $G, $visited, $ans);

for($i=0; $i<count($ans); $i++){
    echo ($i > 0 ? " " : "") . ($ans[$i]+1);
}
echo "\n";

function dfs($v, &$G, &$visited, &$ans, $p=-1){
    $visited[$v] = true;
    array_push($ans, $v);
    foreach($G[$v] as $u){
        if($u == $p || $visited[$u]) continue;
        dfs($u, $G, $visited, $ans, $v);
        array_push($ans, $v);
    }
}

?>
